src/rust/engine/options/src/args.rs

```diff
@@ -228,17 +228,22 @@ impl OptionsSource for Args {
         self.get_list::<String>(id)
     }
 
-    fn get_dict(&self, id: &OptionId) -> Result<Option<DictEdit>, String> {
-        // We iterate in reverse so that the rightmost arg wins in case an option
-        // is specified multiple times.
-        for arg in self.args.iter().rev() {
+    fn get_dict(&self, id: &OptionId) -> Result<Option<Vec<DictEdit>>, String> {
+        let mut edits = vec![];
+        for arg in self.args.iter() {
             if arg.matches(id) {
-                return expand_to_dict(arg.value.clone().ok_or_else(|| {
-                    format!("Expected list option {} to have a value.", self.display(id))
-                })?)
-                .map_err(|e| e.render(&arg.flag));
+                let value = arg.value.clone().ok_or_else(|| {
+                    format!("Expected dict option {} to have a value.", self.display(id))
```

Comment:
I notice you fixed a typo here; should this have been caught by a test? Do we need a test?

Reply:
I'm not sure a test like that would carry its weight, TBH.